federatedscope/attack/auxiliary/create_edgeset.py

```diff
@@ -0,0 +1,141 @@
+
+        # Data Augmentation on images_seven
+        for idx in range(len(images_seven)):
+            for cad_ang in cand_angles:
+                PIL_img = transforms.ToPILImage()(
+                    images_seven[idx]).convert("L")
+                PIL_img_rotate = transforms.functional.rotate(PIL_img,
+                                                              cad_ang,
+                                                              fill=(0, ))
+
+                img_rotate = torch.from_numpy(np.array(PIL_img_rotate))
+                images_seven_DA = torch.cat(
+                    (images_seven_DA,
+                     img_rotate.reshape(1,
+                                        img_rotate.size()[0],
+                                        img_rotate.size()[0])), 0)
+
+                print(images_seven_DA.size())
+
+        poisoned_labels_DA = (torch.zeros(images_seven_DA.size()[0]) +
+                              target_label).long()
+
+    poisoned_edgeset = []
+    if fraction < 1:
+        for ii in range(len(images_seven_cut)):
+            poisoned_edgeset.append(
+                (images_seven_cut[ii], poisoned_labels_cut[ii]))
+
+    else:
+        for ii in range(len(images_seven_DA)):
+            poisoned_edgeset.append(
+                (images_seven_DA[ii], poisoned_labels_DA[ii]))
+    return poisoned_edgeset
+
+
+def create_ardis_test_dataset(data_path, base_label=7, target_label=1):
+
+    # load the data from csv's
+    load_path = data_path + 'ARDIS_test_2828.csv'
+    ardis_images = np.loadtxt(load_path, dtype='float')
+    load_path = data_path + 'ARDIS_test_labels.csv'
+    ardis_labels = np.loadtxt(load_path, dtype='float')
+
+    # reshape to be [samples][height][width]
+    ardis_images = torch.tensor(
+        ardis_images.reshape(ardis_images.shape[0], 28,
+                             28).astype('float32')).type(torch.uint8)
+
+    indices_seven = np.where(ardis_labels[:, base_label] == 1)[0]
+    images_seven = ardis_images[indices_seven, :]
+    images_seven = torch.tensor(images_seven).type(torch.uint8)
+    images_seven = images_seven.unsqueeze(1)
+
+    poisoned_labels = (torch.zeros(images_seven.size()[0]) +
+                       target_label).long()
+    poisoned_labels = torch.tensor(poisoned_labels)
+
+    ardis_test_dataset = []
+
+    for ii in range(len(images_seven)):
+        ardis_test_dataset.append((images_seven[ii], poisoned_labels[ii]))
+
+    return ardis_test_dataset
+
+
+if __name__ == '__main__':
+    '''
+    please use your own path of the data
+    '''
+    data_path = ''
+    fraction = 0.1
+    poisoned_edgeset = create_ardis_poisoned_dataset(fraction=fraction,
+                                                     data_path=data_path)
+
+    ardis_test_dataset = create_ardis_test_dataset(data_path)
+
+    print("Writing poison_data to: ")
+    print("poisoned_edgeset_fraction_{}".format(fraction))
+
+    with open("poisoned_edgeset_fraction_{}".format(fraction),
```

Comment:
All the data processsing file are suggested to save in the directory `cfg.data.root`

Reply:
fix it